workshops/chat/README.md

```diff
@@ -0,0 +1,258 @@
+  ```
+
+## Part II: The HTML File
+
+- Add a div to store message history in the **body**, above the script tags
+
+```html
+<div id="messagesDiv"></div>
+```
+
+- put two inputs into the **body**, above the script tags
+
+```html
+<input type="text" id="nameInput" placeholder="Name">
+<input type="text" id="messageInput" placeholder="Message">
+```
+
+Save and open up the external live preview.
+
+## Part III: The JS File
+
+### Preparing to Use Firebase
+
+First we'll create an app on firebase.
+
+![](img/fb_create_project.gif)
+
+Let's click on "Add Firebase to your web app." We'll copy part of the code snippet and paste it into our `main.js`:
+
+![](img/fb_get_config.gif)
+
+Now we're in the dashboard again.
+
+![](img/TODO dashboard)
+
+We have to add the domain of our external live preview to Firebase, so that we can debug while building the app. We'll click on "Auth" in the sidebar, and then the "Sign-In Method" tab:
+
+![](img/TODO go to side bar "auth")
+![](img/TODO change tab to sign-in method)
+
+![](img/fb_add_domain.gif)
+
+Next we have to open up our database permissions. We'll do this by clicking on "Database" in the sidebar, and editing the aflksdff TODO
+
+![](img/TODO set the shit to true and publicize errthang)
+
+Great. Now we are done configuring Firebase and can return to our JavaScript file.
+
+So far our `main.js` looks like this:
+
+```js
+var config = {
+    apiKey: "AIzaSyCyCLHv4X9TnygLWKoBt52ml_A4TjjNVek",
+    authDomain: "chat-test-bccdb.firebaseapp.com",
+    databaseURL: "https://chat-test-bccdb.firebaseio.com",
+    storageBucket: "",
+};
+firebase.initializeApp(config);
+```
+
+We'll be adding one more line to the bottom of the file. This creates a reference to the database in your Firebase project, so that you can access it within your JavaScript file.
+
+```js
+var myDataRef = firebase.database().ref();
+```
+
+### Saving Messages to Firebase
+
+Now that we've allowed our Firebase project to be accessed by our C9 project, we can perform actions on our Firebase project through our `main.js`.
+
+First, we'll write a function that will push your name and message to the Firebase reference, when you press enter after typing a message.
+
+Let's create the function, and attach it to a keypress on the message input:
+
+```js
+function pushMessage(event) {
+}
+
+$('#messageInput').keypress(pushMessage);
+```
+
+Now we will say "if the event's key code is that of the enter key, do stuff" by adding a conditional. The key code of the enter key is 13, so that's what we will be comparing in the conditional.
+
+```js
+function pushMessage(event) {
+  if (event.keyCode == 13) {
+  }
+}
+```
+
+We need the message's sender and text, so that we can pass them to Firebase. We'll extract the values from the HTML inputs, and store them in variables:
+
+```js
+function pushMessage(event) {
+  if (event.keyCode == 13) {
+    var name = $('#nameInput').val();
+    var text = $('#messageInput').val();
+  }
+}
+```
```

Comment:
Make School uses text tutorials and they have the code for each part hidden under a button labeled "Spoiler." It may not make a huge difference but I remember trying to figure things out myself before actually opening the code up and looking at it. 

It also might not be the greatest solution because of how the workshop is broken down. And it really can't be done in any way with Github.